src/components/EntityHighlighter.js

```diff
@@ -1,213 +0,0 @@
-import React from 'react';
-
-const styles = {
-  text: {},
-  highlightText: {
-    color: 'transparent',
-    pointerEvents: 'none',
-    padding: '0',
-    whiteSpace: 'pre-wrap',
-    fontFamily: 'source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace',
-    fontSize: 14,
-  },
-  zeroPos: {
-    textAlign: 'left',
-    position: 'absolute',
-    top: 1,
-    left: 1,
-  },
-  input: {
-    fontFamily: 'source-code-pro, Menlo, Monaco, Consolas, "Courier New", monospace',
-    fontSize: 14,
-    background: 'none',
-    border: '1px solid',
-    width: '100%',
-    resize: 'none',
-  },
-};
-
-const colors = [
-  { name: 'blue', bg: '#0074d9' },
-  { name: 'navy', bg: '#001f3f' },
-  { name: 'lime', bg: '#01ff70' },
-  { name: 'teal', bg: '#39cccc' },
-  { name: 'olive', bg: '#3d9970' },
-  { name: 'fuchsia', bg: '#f012be' },
-  { name: 'red', bg: '#ff4136' },
-  { name: 'green', bg: '#2ecc40' },
-  { name: 'orange', bg: '#ff851b' },
-  { name: 'maroon', bg: '#85144b' },
-  { name: 'purple', bg: '#b10dc9' },
-  { name: 'yellow', bg: '#ffdc00' },
-  { name: 'aqua', bg: '#7fdbff' },
-];
-
-function hashString(str) {
-  let hash = 0;
-  if (str.length === 0) return hash;
-  for (let i = 0; i < str.length; i += 1) {
-    const char = str.charCodeAt(i);
-    hash = (hash << 5) - hash + char;
-    hash &= hash; // Convert to 32bit integer
-  }
-  return hash > 0 ? hash : -hash;
-}
-
-class EntityHighlighter extends React.Component {
-  constructor(props) {
-    super(props);
-    this.state = { selectionStart: 0, selectionEnd: 0, text: '' };
-  }
-
-  componentDidMount() {
-    this.selectionChangeHandler = (event) => {
-      const target = event.target;
-
-      if (
-        target === this.inputNode
-      ) {
-        this.setState({
-          selectionStart: this.inputNode.selectionStart,
-          selectionEnd: this.inputNode.selectionEnd
-        });
-      }
-    };
-    document.addEventListener('select', this.selectionChangeHandler, false);
-    document.addEventListener('click', this.selectionChangeHandler, false);
-    document.addEventListener('keydown', this.selectionChangeHandler, false);
-  }
-
-  componentWillUnmount() {
-    document.removeEventListener('select', this.selectionChangeHandler);
-    document.removeEventListener('click', this.selectionChangeHandler);
-    document.removeEventListener('keydown', this.selectionChangeHandler);
```

Comment:
Don't access or manipulate the DOM directly when using React, use the functionality provided by the library.